examples/next/src/lib/store/use-provider.ts

```diff
@@ -0,0 +1,193 @@
+      initializing: false,
+
+      // Set the selected chain and its client on user selection (or on first mount)
+      // Fetch the state of the account if there is one
+      setProvider: async (chain, address) => {
+        const {
+          client: currentClient,
+          initializing,
+          initializedClients,
+          forkTime,
+          setForkTime,
+        } = get();
+        // The user should not be able to change the chain while a client is being initialized
+        if (initializing) return null;
+
+        set({ initializing: true });
+
+        // An error might occur, e.g. if trying to connect to a local chain when it's not running
+        try {
+          // 1. Check if we already have the appropriate client for the selected chain
+          // e.g. when searching a different account on the same chain
+          let client =
+            (await currentClient?.getChainId()) === chain.id
+              ? currentClient
+              : null;
+
+          // 2. If not found, try to find the client for the selected chain if it was
+          // already initialized earlier
+          if (!client) {
+            for (const c of initializedClients) {
+              if ((await c.getChainId()) === chain.id) {
+                client = c;
+              }
+            }
+          }
+
+          // 3. If not found, initialize a new client
+          if (!client) {
+            // TODO TEMP await because of lazy import
+            const { createMemoryClient } = await import('tevm');
+            const { createSyncStoragePersister } = await import(
+              'tevm/sync-storage-persister'
+            );
+
+            const forkUrl = chain.custom.rpcUrl;
+
+            client = createMemoryClient({
+              persister: createSyncStoragePersister({
+                storage: localStorage,
+                key: `TEVM_CLIENT_${chain.id.toString()}`,
+              }),
+              fork: {
+                url: STANDALONE_RPC_CHAINS.includes(chain.id)
+                  ? forkUrl
+                  : `${forkUrl}${ALCHEMY_API_KEY}`,
+              },
+            });
+
+            await client.ready();
+
+            // Set its fork time if it's never been initialized
+            // This is aligned with the client being completely new, or already used
+            // before (synced with local storage)
+            if (forkTime[chain.id] === undefined) setForkTime(chain.id);
+
+            // Add the client to the list of initialized clients
+            set({ initializedClients: [...initializedClients, client] });
+          }
+
+          set({ chain, client, initializing: false });
+
+          // 4. If an address is provided, update the state of the account on this chain
+          if (address && client) {
+            useConfigStore.getState().updateAccount(address, {
+              updateAbi: true,
+              chain,
+              client,
+            });
+          }
+
+          return client;
```

Comment:
Would avoid one zustand store updating another zustand store at all costs. In this case I don't think updateAccount needs to exist in zustand at all